h2o-algos/src/main/java/hex/tree/PlattScalingHelper.java

```diff
@@ -0,0 +1,124 @@
+        GLMModel calibrationModel();
+    }
+    
+    public static void initCalibration(ModelBuilderWithCalibration builder, ParamsWithCalibration parms, boolean expensive) {
+        // Calibration
+        Frame cf = parms.getCalibrationFrame();  // User-given calibration set
+        if (cf != null) {
+            if (! parms.calibrateModel())
+                builder.warn("_calibration_frame", "Calibration frame was specified but calibration was not requested.");
+            Frame adaptedCf = builder.init_adaptFrameToTrain(cf, "Calibration Frame", "_calibration_frame", expensive);
+            builder.setCalibrationFrame(adaptedCf);
+        }
+        if (parms.calibrateModel()) {
+            if (builder.nclasses() != 2)
+                builder.error("_calibrate_model", "Model calibration is only currently supported for binomial models.");
+            if (cf == null)
+                builder.error("_calibrate_model", "Calibration frame was not specified.");
+        }
+    }
+    
+    public static <M extends Model<M , P, O>, P extends Model.Parameters, O extends Model.Output> void buildCalibrationModel(
+        boolean finalScoring, ModelBuilderWithCalibration<M, P, O> builder, 
+        ParamsWithCalibration parms, Job job, M model
+    ) {
+        // Model Calibration (only for the final model, not CV models)
+        if (finalScoring && parms.calibrateModel() && (!parms.isCVModel())) {
+            Key<Frame> calibInputKey = Key.make();
+            try {
+                Scope.enter();
+                job.update(0, "Calibrating probabilities");
+                Frame calib = builder.getCalibrationFrame();
+                Vec calibWeights = parms.weightsColumn() != null ? calib.vec(parms.weightsColumn()) : null;
+                Frame calibPredict = Scope.track(model.score(calib, null, job, false));
+                Frame calibInput = new Frame(calibInputKey,
+                    new String[]{"p", "response"}, new Vec[]{calibPredict.vec(1), calib.vec(parms.responseColumn())});
+                if (calibWeights != null) {
+                    calibInput.add("weights", calibWeights);
+                }
+                DKV.put(calibInput);
+
+                Key<Model> calibModelKey = Key.make();
+                Job calibJob = new Job<>(calibModelKey, ModelBuilder.javaName("glm"), "Platt Scaling (GLM)");
+                GLM calibBuilder = ModelBuilder.make("GLM", calibJob, calibModelKey);
+                calibBuilder._parms._intercept = true;
+                calibBuilder._parms._response_column = "response";
+                calibBuilder._parms._train = calibInput._key;
+                calibBuilder._parms._family = GLMModel.GLMParameters.Family.binomial;
+                calibBuilder._parms._lambda = new double[] {0.0};
+                if (calibWeights != null) {
+                    calibBuilder._parms._weights_column = "weights";
+                }
+
+                builder.setCalibrationOutput(calibBuilder, model);
```

Comment:
Can this method instead return the calibration model? I think it is a little unexpected that responsibility of this method is also to set it to the original builder.

Reply:
done